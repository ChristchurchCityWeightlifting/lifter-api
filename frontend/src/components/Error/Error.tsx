import React from "react";
import Alert from "@mui/material/Alert";
import Box from "@mui/material/Box";

const Error: React.FC = () => {
  return (
    <Box sx={{ width: "100%", height: "40%" }}>
      <Alert severity="error"> Error </Alert>
    </Box>
  );
};

export default Error;
