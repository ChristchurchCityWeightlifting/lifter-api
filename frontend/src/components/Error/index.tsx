/** @format */

export { default } from "./Error";
