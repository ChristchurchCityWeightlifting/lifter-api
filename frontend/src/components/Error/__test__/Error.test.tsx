/** @format */

import React from "react";
import { describe, test, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import Error from "../Error";

describe("Error components tests", () => {
  test("renders", () => {
    const tree = renderer.create(<Error />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
