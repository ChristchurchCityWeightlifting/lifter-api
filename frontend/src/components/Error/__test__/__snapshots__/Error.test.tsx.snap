// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error components tests renders 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1fuhjh6{max-width:-webkit-fit-content;max-width:-moz-fit-content;max-width:fit-content;}",
      }
    }
    data-emotion="css 1fuhjh6"
  />,
  <div
    className="MuiBox-root css-1fuhjh6"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-ek4erk-MuiAlert-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;background-color:rgb(253, 237, 237);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:6px 16px;color:rgb(95, 33, 32);}.css-ek4erk-MuiAlert-root .MuiAlert-icon{color:#ef5350;}",
        }
      }
      data-emotion="css ek4erk-MuiAlert-root"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-xilmwd-MuiPaper-root-MuiAlert-root{background-color:#fff;color:rgba(0, 0, 0, 0.87);-webkit-transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border-radius:4px;box-shadow:none;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;background-color:rgb(253, 237, 237);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:6px 16px;color:rgb(95, 33, 32);}.css-xilmwd-MuiPaper-root-MuiAlert-root .MuiAlert-icon{color:#ef5350;}",
        }
      }
      data-emotion="css xilmwd-MuiPaper-root-MuiAlert-root"
    />
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-xilmwd-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1ytlwq5-MuiAlert-icon{margin-right:12px;padding:7px 0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;font-size:22px;opacity:0.9;}",
          }
        }
        data-emotion="css 1ytlwq5-MuiAlert-icon"
      />
      <div
        className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-j7qwjs{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
            }
          }
          data-emotion="css j7qwjs"
        />
        <div
          className="MuiBox-root css-j7qwjs"
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1vooibu-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:inherit;}",
              }
            }
            data-emotion="css 1vooibu-MuiSvgIcon-root"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="FluorescentIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 9h14v6H5zm6-7h2v3h-2zm6.2863 4.3989 1.7897-1.8024 1.4192 1.4092-1.7897 1.8024zM11 19h2v3h-2zm6.29-1.29 1.79 1.8 1.42-1.42-1.8-1.79zM3.4947 6.0091 4.902 4.602l1.789 1.789-1.4074 1.407zm-.0026 12.0672 1.8024-1.7897 1.4092 1.4192-1.8023 1.7897z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="FluorescentIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 9h14v6H5zm6-7h2v3h-2zm6.2863 4.3989 1.7897-1.8024 1.4192 1.4092-1.7897 1.8024zM11 19h2v3h-2zm6.29-1.29 1.79 1.8 1.42-1.42-1.8-1.79zM3.4947 6.0091 4.902 4.602l1.789 1.789-1.4074 1.407zm-.0026 12.0672 1.8024-1.7897 1.4092 1.4192-1.8023 1.7897z"
            />
          </svg>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-16o2g24-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:inherit;-webkit-animation:blink-animation;animation:blink-animation;}",
              }
            }
            data-emotion="css 16o2g24-MuiSvgIcon-root"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-16o2g24-MuiSvgIcon-root"
            data-testid="FluorescentIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 9h14v6H5zm6-7h2v3h-2zm6.2863 4.3989 1.7897-1.8024 1.4192 1.4092-1.7897 1.8024zM11 19h2v3h-2zm6.29-1.29 1.79 1.8 1.42-1.42-1.8-1.79zM3.4947 6.0091 4.902 4.602l1.789 1.789-1.4074 1.407zm-.0026 12.0672 1.8024-1.7897 1.4092 1.4192-1.8023 1.7897z"
            />
          </svg>
        </div>
      </div>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1pxa9xg-MuiAlert-message{padding:8px 0;min-width:0;overflow:auto;}",
          }
        }
        data-emotion="css 1pxa9xg-MuiAlert-message"
      />
      <div
        className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-zwr1uk-MuiAlertTitle-root{font-weight:500;margin-top:-2px;}",
            }
          }
          data-emotion="css zwr1uk-MuiAlertTitle-root"
        />
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-yslrua-MuiTypography-root-MuiAlertTitle-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;margin-bottom:0.35em;font-weight:500;margin-top:-2px;}",
            }
          }
          data-emotion="css yslrua-MuiTypography-root-MuiAlertTitle-root"
        />
        <div
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
        >
           Three Red Lights
        </div>
        Something went wrong. Please try again later.
      </div>
    </div>
  </div>,
]
`;
